superset/daos/key_value.py

```diff
@@ -0,0 +1,145 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import logging
+from datetime import datetime
+from typing import Any
+from uuid import UUID
+
+from sqlalchemy import and_
+
+from superset import db
+from superset.daos.base import BaseDAO
+from superset.key_value.exceptions import (
+    KeyValueCreateFailedError,
+    KeyValueUpdateFailedError,
+)
+from superset.key_value.models import KeyValueEntry
+from superset.key_value.types import Key, KeyValueCodec, KeyValueResource
+from superset.key_value.utils import get_filter
+from superset.utils.core import get_user_id
+
+logger = logging.getLogger(__name__)
+
+
+class KeyValueDAO(BaseDAO[KeyValueEntry]):
+    @staticmethod
+    def get_entry(
```

Comment:
To reviewers who think the `_entry` suffix seems redundant: `BaseDAO` already has `create`, `update`, `delete`, which expects passing an instance of the model. This doesn't fit this use case very well, as the interface here is slightly different from the typical ORM object:
- `key` can be an `int` or `UUID`
- The Key Value model encodes/decodes values using a codec that we want to enforce in our method signatures

For this reason I needed to name these slightly differently to not collide with the base methods.